### PR TITLE
IALERT-3315 saml metadata mode

### PR DESCRIPTION
**Pull request by mau-synopsys**
Dependent on https://github.com/blackducksoftware/blackduck-alert/pull/2329

-Create a table for saml metadata mode
-Create enum for saml metadata mode
-Create converter to convert mode(backend) to int(database)
-Added metadata mode to model and entity

### CODE REVIEW

**Review by bseifert14 — APPROVED** [review]

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/configuration/SAMLConfigurationEntity.java [hunk 1]
```diff
@@ -64,19 +77,17 @@ public SAMLConfigurationEntity(
...
 
-    public String getName() {
-        return name;
-    }
+    public String getName() { return name; }
 
-    public OffsetDateTime getCreatedAt() {
-        return createdAt;
-    }
+    public OffsetDateTime getCreatedAt() { return createdAt; }
```
Comment (ChomickiM) [inline authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/configuration/SAMLConfigurationEntity.java:90]: minor thing, but the returns should be on their own line

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/SAMLMetaDataModeConverter.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.synopsys.integration.alert.authentication.saml.database;
+
+import com.synopsys.integration.alert.authentication.saml.model.SAMLMetadataMode;
+
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+import java.util.stream.Stream;
+
+@Converter(autoApply = true)
+public class SAMLMetaDataModeConverter implements AttributeConverter<SAMLMetadataMode, Integer> {
```
Comment (psantos1113) [inline authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/SAMLMetaDataModeConverter.java:10]: We should probably do this for any other enums that we save to the database.

**Review by psantos1113 — APPROVED** [review]

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/configuration/SAMLConfigurationEntity.java (hunk 1 above)
Comment (mau-synopsys) [reply authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/configuration/SAMLConfigurationEntity.java]: Changed to multi-line

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/SAMLMetaDataModeConverter.java (hunk 2 above)
Comment (mau-synopsys) [reply authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/database/SAMLMetaDataModeConverter.java:10]: Created https://jira-sig.internal.synopsys.com/browse/IALERT-3320

**Pull request merged**